apps/token/src/routes/governance/propose/update-asset/propose-update-asset.tsx

```diff
@@ -0,0 +1,190 @@
+  ProposalFormDescription,
+  ProposalFormTerms,
+  ProposalFormSubmit,
+  ProposalFormTransactionDialog,
+  ProposalFormSubheader,
+  ProposalFormVoteAndEnactmentDeadline,
+} from '../../components/propose';
+import { AsyncRenderer, Link } from '@vegaprotocol/ui-toolkit';
+import { Heading } from '../../../../components/heading';
+import { VegaWalletContainer } from '../../../../components/vega-wallet-container';
+import { NetworkParams, useNetworkParams } from '@vegaprotocol/react-helpers';
+
+export interface UpdateAssetProposalFormFields {
+  proposalVoteDeadline: string;
+  proposalEnactmentDeadline: string;
+  proposalTitle: string;
+  proposalDescription: string;
+  proposalTerms: string;
+  proposalReference: string;
+}
+
+const docsLink = '/update-asset-proposal';
+
+export const ProposeUpdateAsset = () => {
+  const {
+    params,
+    loading: networkParamsLoading,
+    error: networkParamsError,
+  } = useNetworkParams([
+    NetworkParams.governance_proposal_updateAsset_minClose,
+    NetworkParams.governance_proposal_updateAsset_maxClose,
+    NetworkParams.governance_proposal_updateAsset_minEnact,
+    NetworkParams.governance_proposal_updateAsset_maxEnact,
+    NetworkParams.governance_proposal_updateAsset_minProposerBalance,
+    NetworkParams.spam_protection_proposal_min_tokens,
+  ]);
+
+  const { VEGA_EXPLORER_URL, VEGA_DOCS_URL } = useEnvironment();
+  const { t } = useTranslation();
+  const {
+    register,
+    handleSubmit,
+    formState: { isSubmitting, errors },
+  } = useForm<UpdateAssetProposalFormFields>();
+  const { finalizedProposal, submit, Dialog } = useProposalSubmit();
+
+  const onSubmit = async (fields: UpdateAssetProposalFormFields) => {
+    await submit({
+      rationale: {
+        title: fields.proposalTitle,
+        description: fields.proposalDescription,
+      },
+      terms: {
+        updateAsset: {
+          ...JSON.parse(fields.proposalTerms),
+        },
+        closingTimestamp: getClosingTimestamp(fields.proposalVoteDeadline),
+        enactmentTimestamp: getEnactmentTimestamp(
+          fields.proposalVoteDeadline,
+          fields.proposalEnactmentDeadline
+        ),
+      },
+    });
+  };
+
+  return (
+    <AsyncRenderer
+      loading={networkParamsLoading}
+      error={networkParamsError}
+      data={params}
+    >
+      <Heading title={t('UpdateAssetProposal')} />
+      <VegaWalletContainer>
+        {() => (
+          <>
+            <ProposalFormMinRequirements
+              minProposerBalance={
+                params.governance_proposal_updateAsset_minProposerBalance
+              }
+              spamProtectionMin={params.spam_protection_proposal_min_tokens}
+            />
+
+            {VEGA_DOCS_URL && (
+              <p className="text-sm" data-testid="proposal-docs-link">
+                <span className="mr-1">{t('ProposalTermsText')}</span>
+                <Link
+                  href={`${VEGA_DOCS_URL}/tutorials/proposals${docsLink}`}
```

Comment:
can we pull this part out into a const as well please as it is duplicated here and a couple lines down

Reply:
This will be sorted by #1607